rust/lance/src/dataset/scanner.rs

```diff
@@ -181,6 +181,54 @@ impl MaterializationStyle {
     }
 }
 
+/// Filter for filtering rows
+pub enum LanceFilter {
+    /// The filter is an SQL string
+    Sql(String),
+    /// The filter is a Substrait expression
+    Substrait(Vec<u8>),
+    /// The filter is a Datafusion expression
+    Datafusion(Expr),
+}
+
+impl LanceFilter {
+    /// Converts the filter to a Datafusion expression
+    ///
+    /// The schema for this conversion should be the full schema available to
+    /// the filter (`full_schema`).  However, due to a limitation in the way
+    /// we do Substrait conversion today we can only do Substrait conversion with
+    /// the dataset schema (`dataset_schema`).  This means that Substrait will
+    /// not be able to access columns that are not in the dataset schema (e.g.
+    /// _rowid, _rowaddr, etc.)
+    #[allow(unused)]
```

Comment:
This was needed because `dataset_schema` is unused if the `substrait` feature is not specified.